docs/developer-guide/firstanalysis.md

```diff
@@ -0,0 +1,144 @@
+# Your first analysis
+
+This is a tutorial exercise for writing analyses in Goblint.
+We will implement a very simple sign analysis.
+You need to install goblint for development.
+
+## First test
+
+Create a C file in the Goblint root directory for testing.
+Here we call it `example.c`. This is the first test:
+
+
+```
+#include<assert.h>
+
+int main() {
+
+  int x;
+  int unknown;
+
+  if (unknown) {
+    x = 5;
+  } else {
+    x = 7;
+  }
+
+  // The above code branches on an uninitialized variable.
+  // The value of x could be either 5 or 7.
+
+  assert(x > 0);  // Thus, this assertion should hold!
+
+  return 0;
+}
+```
+
+If you run goblint out of the box on this example it will not work: `./goblint --enable dbg.debug example.c` will claim that the assertion in unknown.
+Goblint could verify that this asertion does hold using interval analysis (`--enable ana.int.interval`), but here we will implement a simple sign analysis instead.
+
+We begin by registering and running a new analysis called "signs":
+
+1. Create a copy of the unit analysis (*src/analyses/unit.ml*) into the file *src/analyses/signs.ml*.
+2. Inside the file, change the name from "unit" to "signs".
+3. Try to run goblint with the new analysis enabled: `./goblint --sets "ana.activated[+]" signs --enable dbg.debug example.c`. The result will still be that nothing is verified.
```

Comment:
No need for `""` around `ana.activated[+]`.

Reply:
They're needed when using `zsh`.

Reply:
Thanks for pointing that out. Make sense to leave it as is then. I'm now a bit worried that there could be other places where it's been used without quotes, but if they ever come up they can be fixed. Most of our scripts use a bash shebang so I guess that also has avoided such issues from appearing.